synapse/config/server.py

```diff
@@ -410,6 +410,13 @@ def read_config(self, config, **kwargs):
             ["0.0.0.0", "::"],
             config_path=("federation_ip_range_blacklist",),
         )
+        # The federation_ip_range_whitelist is used for backwards-compatibility
+        # and will always be None, as it was never set in any configuration files.
+        # If no backwards-compatibility is required, i.e. the configuration file
+        # doesn't set federation_ip_range_blacklist, use ip_range_whitelist instead.
```

Comment:
Hrm yes, this is still a bit confusing.

How about something like this:

```python
        # federation_ip_range_blacklist is used for backwards-compatibility
        # and only applies to federation and identity servers.
        if "federation_ip_range_blacklist" in config:
            # Always blacklist 0.0.0.0, ::
            self.federation_ip_range_blacklist = generate_ip_set(
                config["federation_ip_range_blacklist"],
                ["0.0.0.0", "::"],
                config_path=("federation_ip_range_blacklist",),
            )
            # 'federation_ip_range_whitelist' was never a supported configuration option.
            self.federation_ip_range_whitelist = None
        else:
            # federation_ip_range_blacklist is not given.
            # Default to ip_range_blacklist and ip_range_whitelist.
            self.federation_ip_range_blacklist = self.ip_range_blacklist
            self.federation_ip_range_whitelist = self.ip_range_whitelist
```

To me, that feels like makes the logic clear without needing any extra explanation. @mikure , @clokep wdyt?

Reply:
That seems quite a bit more explicit, I think it is clearer. 👍

Reply:
I agree. That makes is much clearer. I'll test and update later, when I have more time.

Reply:
I just finished testing and I couldn't find any combinations, that didn't work as I expected.
It's also easier to not miss any combinations, as the code is much clearer now. Thank you for this suggestion, @richvdh.